src/fides/api/ops/service/connectors/dynamodb_connector.py

```diff
@@ -0,0 +1,135 @@
+from fides.api.ops.util.logger import Pii
+from fides.connectors.models import (
+    AWSConfig,
+    ConnectorAuthFailureException,
+    ConnectorFailureException,
+)
+
+
+class DynamoDBConnector(BaseConnector[Any]):  # type: ignore
+    """AWS DynamoDB Connector"""
+
+    def build_uri(self) -> None:
+        """Not used for this type"""
+
+    def close(self) -> None:
+        """Not used for this type"""
+
+    def create_client(self) -> Any:  # type: ignore
+        """Returns a client for a DynamoDB instance"""
+        config = DynamoDBSchema(**self.configuration.secrets or {})
+        try:
+            aws_config = AWSConfig(
+                region_name=config.region_name,
+                aws_access_key_id=config.aws_access_key_id,
+                aws_secret_access_key=config.aws_secret_access_key,
+            )
+            return aws_connector.get_aws_client(
+                service="dynamodb", aws_config=aws_config
+            )
+        except ValueError:
+            raise ConnectionException("Value Error connecting to AWS DynamoDB.")
+
+    def query_config(self, node: TraversalNode) -> QueryConfig[Any]:
+        """Query wrapper corresponding to the input traversal_node."""
+        return DynamoDBQueryConfig(node)
+
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        """
+        Connects to AWS DynamoDB and lists tables to validate credentials.
+        """
+        logger.info("Starting test connection to {}", self.configuration.key)
+        client = self.client()
+        try:
+            client.list_tables()
+        except ClientError as error:
+            if error.response["Error"]["Code"] in [
+                "InvalidClientTokenId",
+                "SignatureDoesNotMatch",
+            ]:
+                raise ConnectorAuthFailureException(error.response["Error"]["Message"])
+            raise ConnectorFailureException(error.response["Error"]["Message"])
+
+        return ConnectionTestStatus.succeeded
+
+    def retrieve_data(
+        self,
+        node: TraversalNode,
+        policy: Policy,
+        privacy_request: PrivacyRequest,
+        input_data: Dict[str, List[Any]],
+    ) -> List[Row]:
+        """Retrieve DynamoDB data"""
+        query_config = self.query_config(node)
+        collection_name = node.address.collection
+        client = self.client()
+        try:
+            describe_table = client.describe_table(TableName=collection_name)
+            input_data["attribute_definitions"] = describe_table["Table"][
+                "AttributeDefinitions"
+            ]
+            query_param = query_config.generate_query(input_data, policy)
+            item = client.get_item(
+                TableName=collection_name,
+                Key=query_param,  # type: ignore
+            )
+            result = {}
+            if "Item" in item:
+                for key in item["Item"]:
+                    result[key] = [*item["Item"][key].values()][0]
```

Comment:
We should check for instances where the value is a list or a complex type. For example:
```
{
  "book_id": 1,
  "title": "Sample Book",
  "author": "John Doe",
  "genres": {"Mystery", "Thriller"},
  "publish_date": "2022-01-01",
  "ratings": [
    {
      "user_id": 101,
      "rating": 4
    },
    {
      "user_id": 102,
      "rating": 5
    }
  ]
}
```
We might miss some data if we only take the first element

Reply:
That makes sense 👍🏽 thanks @galvana ! [Data types listed here](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes) so it is definitely a thing.

Do you know if there are any bounds to the scenarios we are trying to account for? i.e. I'm trying to think of if we currently support referencing a list of IDs instead of a single item or if it is primarily attributes